internal/services/instances.go

```diff
@@ -534,3 +536,25 @@ func convertUnstructuredListIntoHelmCharts(unstructuredList *unstructured.Unstru
 
 	return helmChartList, nil
 }
+
+func concatenateCatalogServiceValues(catalogService models.CatalogService, name string) string {
+
+	type FixedValues struct {
+		ServiceName string `json:"serviceName,omitempty"`
+	}
+	var fixedValues FixedValues
+	var builder strings.Builder
+
+	fixedValues.ServiceName = name
+	// Build the concatenated string
+
+	builder.WriteString("\nserviceName: ")
+	builder.WriteString(fixedValues.ServiceName)
+	builder.WriteString("\n")
+
+	// Assign the result to catalogService.Values
+	catalogService.Values += builder.String()
```

Comment:
I am not sure why you are adding this to `catalogService.Values` here, and return `catalogService.Values`, and then in the caller it is assigned to `catalogService.Values` again.

Reply:
I would simply return the `builder.String()` result and do the `+=` in the caller.

Reply:
And with the YAML generation confirmed I would return the string conversion of the `yaml.Marshall` result instead.